### ISSUE TITLE
QA: Add a code formatter and code linter for standardizing the code style

### ISSUE THREAD

**Issue by Rahulm2310** [anitab-org/bridge-in-tech-web#174]
**Is your feature request related to a problem? Please describe.**
It will be difficult to enforce a coding style as the repo grows bigger and gets more PRs.

**Describe the solution you'd like**
We must use a code formatter (ideally, opinionated one like https://prettier.io/) and a code linter (like https://eslint.org/) to enhance code quality and maintain consistency.

**Comment by Rahulm2310** [comment]
@mtreacy002 @isabelcosta  Is this valid? If yes, I would like to work on this. Thanks 🙂

**Comment by mtreacy002** [comment]
@Rahulm2310 . I think this is a valid issue. Can you please tell me which of the 2 you mentioned above you're going to use? Can you please share your insight on pros and cons on both options here? Thanks

**Comment by Rahulm2310** [comment]
![image](https://user-images.githubusercontent.com/54268438/108104017-c36a2300-70b0-11eb-8c87-c6cac85c7548.png)

@mtreacy002 We can use integrate both of them to get the best results (a new trend). Prettier also provides a [guide](https://prettier.io/docs/en/integrating-with-linters.html) to integrate with eslint.

**Comment by mtreacy002** [comment]
Sounds good to me (to integrate both), @Rahulm2310 . The next question is, how and when will we apply the linters? Can they be added to the CI build (currently we're using Travis but soon moving to Github action) which is an ideal way so no need for manual linting.

**Comment by Rahulm2310** [comment]
Yes @mtreacy002 . Both of these can be added to Github Actions to avoid manual linting. But should we put this on hold till we use Travis ?

**Comment by mtreacy002** [comment]
Definitely, @Rahulm2310. In fact, we're moving to Github Actions, so once #170 approved and merged, you can work on this issue and add the linters as GitHub workflow 😉

**Comment by Rahulm2310** [comment]
Sure. Thanks @mtreacy002  😉

**Comment by mtreacy002** [comment]
@Rahulm2310 , I forgot that there's a [black linter issue](https://github.com/anitab-org/bridge-in-tech-backend/issues/184) that will address this as part of github workfkow. I'll close this issue now to avoid duplication 😉

**Comment by Rahulm2310** [comment]
@mtreacy002 Isn't that for backend? We need a linter for frontend as well

**Comment by mtreacy002** [comment]
Oops, you're right, @Rahulm2310 . Sorry for the confusion. Reopening this again 😊.

**Comment by satyaarthchhabra** [comment]
hi @mtreacy . can i work on this issue

**Comment by vj-codes** [comment]
@satyaarthchhabra hey are you still interested to work on this?

**Comment by ritik307** [comment]
@vj-codes I would like yo work on this issue

**Comment by vj-codes** [comment]
Assigning you @ritik307 
Happy coding!

**Comment by ritik307** [comment]
@vj-codes I have made a PR regarding the issue but the build is failing bcz of the below-mentioned issue. The issue is forcing me to install an older version of eslint which in my perspective should not be so.
Kindly look into the issue.
![error](https://user-images.githubusercontent.com/22374829/147875169-3f5f2630-e8d0-47e5-89ac-78f754c6ad83.png)
